x-pack/legacy/plugins/triggers_actions_ui/np_ready/public/application/components/builtin_action_types/types.ts

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import { ActionConnector } from '../../../types';
+
+export interface EmailActionParams {
+  to: string[];
+  cc: string[];
+  bcc: string[];
+  subject: string;
+  message: string;
+}
+
+export enum EventActionOptions {
+  TRIGGER = 'trigger',
+  RESOLVE = 'resolve',
+  ACKNOWLEDGE = 'acknowledge',
+}
+
+export enum SeverityActionOptions {
+  CRITICAL = 'critical',
+  ERROR = 'error',
+  WARNING = 'warning',
+  INFO = 'info',
+}
+
+export interface PagerDutyActionParams {
+  eventAction?: EventActionOptions;
+  dedupKey?: string;
+  summary?: string;
+  source?: string;
+  severity?: SeverityActionOptions;
+  timestamp?: string;
+  component?: string;
+  group?: string;
+  class?: string;
+}
+
+export interface IndexActionParams {
+  index?: string;
+  refresh?: boolean;
+  executionTimeField?: string;
+  documents?: string[];
```

Comment:
technically, `documents` could be non-optional - `documents: string[]` - as is done with `email`'s `to`/`cc`/`bcc`.  I guess any of the params that are arrays could be done as non-optional like that, with a zero-length array meaning the same as if the property (eg `documents`) was `undefined`. 

Doesn't matter too much, I think, whatever is easier to handle in the code.